frontend/src/components/card/moment/audioCard.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+
+import { IonCol, IonButton, IonIcon, IonGrid, IonRow } from '@ionic/react'
+
+import { playCircle, pauseCircle } from 'ionicons/icons'
+
+export interface AudioVariant {
+  variant: 'audio'
+  audio: string
+}
+export const AudioVariantCard: React.FC<AudioVariant> = props => {
+  const [play, setPlay] = React.useState(true)
+  const clickButton = () => {
+    setPlay(e => !e)
+  }
+  return (
+    <IonGrid>
+      <IonRow className="ion-align-items-center">
+        <IonCol></IonCol>
+        <IonCol>
+          <IonButton size="large" fill="clear" onClick={clickButton}>
```

Comment:
```suggestion
          <IonButton size="large" fill="clear" onClick={setPlay(!play)}>
```

Reply:
`setPlay(!play)` is not the correct way to update the State when the new value is based on the current state. This can cause to some issues while rendering ore in computation the state.

Update the State based on the current Value:
https://reactjs.org/docs/hooks-reference.html#functional-updates

The second Problem here is, the Function is invoked immediately and can not be invoked again because it is not wrapped inside a anonymous function.

This solution works:
```suggestion
     <IonButton size="large" fill="clear" onClick={() => {setPlay(e => !e)}}>
```
but in my opinion its hard to read compare to this solution, with the logic define in a separate function
```suggestion
     <IonButton size="large" fill="clear" onClick={clickButton}>
```

punish me if i am wrong :)

Reply:
> setPlay(!play) is not the correct way to update the State when the new value is based on the current state. This can cause to some issues while rendering or in computation the state.

Read the documentation and other articles the "based" is referenced as doing calculations with it and not setting it. I mean you could do it instead of `setPlay(!play)` -> `setPlay(true)`. `!play` is only a referenced with is either `true` or `false` and not nothing you do with. 

> The second Problem here is, the Function is invoked immediately and can not be invoked again because it is not wrapped inside a anonymous function.

It can be toggled/invoked multiple times. Give it a try.

Reply:
https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous